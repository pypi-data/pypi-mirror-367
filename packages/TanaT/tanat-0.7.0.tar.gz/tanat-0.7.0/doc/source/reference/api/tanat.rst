tanat package
=============

Subpackages
-----------

.. toctree::
   :maxdepth: 4

   tanat.clustering
   tanat.criterion
   tanat.dataset
   tanat.function
   tanat.loader
   tanat.metric
   tanat.mixin
   tanat.runner
   tanat.sequence
   tanat.survival
   tanat.trajectory
   tanat.utils
   tanat.visualization
   tanat.zeroing

Submodules
----------

tanat.cli module
----------------

.. automodule:: tanat.cli
   :members:
   :undoc-members:
   :show-inheritance:

tanat.exception module
----------------------

.. automodule:: tanat.exception
   :members:
   :undoc-members:
   :show-inheritance:

tanat.version module
--------------------

.. automodule:: tanat.version
   :members:
   :undoc-members:
   :show-inheritance:

Module contents
---------------

.. automodule:: tanat
   :members:
   :undoc-members:
   :show-inheritance:
