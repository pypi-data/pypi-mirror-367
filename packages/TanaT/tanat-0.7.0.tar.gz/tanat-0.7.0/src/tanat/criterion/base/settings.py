#!/usr/bin/env python3
"""
Base class for criterion.
"""

import dataclasses

from .enum import CriterionLevel


@dataclasses.dataclass
class Criterion:
    """
    Base class for criterion.
    """

    _REGISTER_NAME = None

    def get_compatibility_levels(self, max_level=CriterionLevel.TRAJECTORY):
        """List the compatibility levels of the criterion.
        Compatibility levels are marked with decorators.

        Args:
            max_level (CriterionLevel, optional): The maximum level to consider.
                Defaults to Trajectory. Entity < Sequence < Trajectory

        Returns a list of CriterionLevel representing the compatibility levels of the
        criterion. The levels are "sequence","entity" or "trajectory".
        """
        if isinstance(max_level, str):
            max_level = CriterionLevel.from_str(max_level)

        compatibility_levels = getattr(self, "COMPATIBILITY_LEVELS", set())
        return [level for level in compatibility_levels if level <= max_level]
