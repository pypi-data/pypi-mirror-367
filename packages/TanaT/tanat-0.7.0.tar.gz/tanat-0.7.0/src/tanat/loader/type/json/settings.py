#!/usr/bin/env python3
"""
Loader for JSON files.
"""

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer
from pypassist.fallback.typing import Dict


@viewer
@dataclass
class JSONLoaderSettings:
    """
    Settings for JSON loader

    Attributes:
        filepath: Path to the JSON file to load
        encoding: File encoding (utf-8 by default)
        json_load_kwargs: Additional arguments to pass to json.load
    """

    filepath: str
    encoding: str = "utf-8"
    json_load_kwargs: Dict = None

    def __post_init__(self):
        self.json_load_kwargs = self.json_load_kwargs or {}
