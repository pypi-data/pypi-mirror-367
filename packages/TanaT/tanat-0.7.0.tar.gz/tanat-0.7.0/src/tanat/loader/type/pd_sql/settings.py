#!/usr/bin/env python3
"""
Dataframe Loader Settings for SQL databases.
"""

from typing import Optional

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer
from pypassist.fallback.typing import Dict


@viewer
@dataclass
class PandasSQLLoaderSettings:
    """
    Settings for SQL loader

    Attributes:
        conn: SQLAlchemy connection string (e.g., 'sqlite:///database.db')
        query: SQL query to execute
        pd_read_sql_kwargs: Additional arguments to pass to pd.read_sql
    """

    conn: str
    query: str
    pd_read_sql_kwargs: Optional[Dict] = None

    def __post_init__(self):
        self.pd_read_sql_kwargs = self.pd_read_sql_kwargs or {}
