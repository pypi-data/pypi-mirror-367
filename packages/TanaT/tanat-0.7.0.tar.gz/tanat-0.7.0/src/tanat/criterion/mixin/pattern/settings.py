#!/usr/bin/env python3
"""
Pattern criterion.
"""

from typing import Union

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer
from pypassist.fallback.typing import Dict, List

from ...base.settings import Criterion
from ...decorator.entity import entity_compatible
from ...decorator.sequence import sequence_compatible
from .enum import PatternOperator


@viewer
@entity_compatible
@sequence_compatible
@dataclass
class PatternCriterion(Criterion):
    """
    Flexible pattern-based filtering for sequences and entities.

    Enables complex pattern matching across data columns with advanced
    configuration options.

    Args:
        pattern (Dict[str, Union[str, List[str]]]):
            Column patterns to match. Supports:
            - Plain strings
            - Lists of strings
            - Regex patterns (prefix with 'regex:')

        contains (bool, optional):
            - True: Include sequences matching pattern
            - False: Exclude sequences matching pattern
            Defaults to True.

        case_sensitive (bool, optional):
            Whether matching is case-sensitive. Defaults to True.

        operator (PatternOperator, optional):
            Logical operator between patterns:
            - AND: All patterns must match
            - OR: At least one pattern matches
            Defaults to AND.

    Example:
        # Find sequences with emergency events
        PatternCriterion(pattern={'event_type': 'EMERGENCY'})
    """

    pattern: Dict[str, Union[str, List[str]]]
    contains: bool = True
    case_sensitive: bool = True
    operator: PatternOperator = PatternOperator.AND

    _REGISTER_NAME = "pattern"
