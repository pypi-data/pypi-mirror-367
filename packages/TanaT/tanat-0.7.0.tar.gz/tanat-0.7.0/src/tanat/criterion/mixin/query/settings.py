#!/usr/bin/env python3
"""
Settings for query criterion.
"""

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer

from ...base.settings import Criterion
from ...decorator.entity import entity_compatible
from ...decorator.sequence import sequence_compatible


@viewer
@entity_compatible
@sequence_compatible
@dataclass
class QueryCriterion(Criterion):
    """
    Pandas-based filtering criterion for sequences and entities.

    Allows complex data filtering using pandas query syntax. Compatible with
    sequence and entity levels, enabling flexible data selection based on
    column conditions.

    Args:
        query (str): Pandas-style query string to filter data.
            Supports complex boolean conditions across columns.

    Example:
        # Filter sequences where age > 50 and gender is 'M'
        criterion = QueryCriterion(query="age > 50 and gender == 'M'")
    """

    query: str

    _REGISTER_NAME = "query"
