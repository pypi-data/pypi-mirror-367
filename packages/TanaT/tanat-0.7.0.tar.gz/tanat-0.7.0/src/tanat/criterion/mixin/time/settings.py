#!/usr/bin/env python3
"""
Settings for time-based filtering.
"""

from typing import Optional, Union
from datetime import datetime

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer

from ...base.settings import Criterion
from ...decorator.entity import entity_compatible
from ...decorator.sequence import sequence_compatible


@viewer
@entity_compatible
@sequence_compatible
@dataclass
class TimeCriterion(Criterion):
    """
    Flexible time-based filtering for sequences and entities.

    Enables precise temporal filtering with multiple time constraints and
    different containment modes depending on the filtering level.

    Args:
        start_before (datetime, optional):
            Filter sequences or entities starting before this timestamp.
        start_after (datetime, optional):
            Filter sequences or entities starting after this timestamp.
        end_before (datetime, optional):
            Filter sequences or entities ending before this timestamp.
        end_after (datetime, optional):
            Filter sequences or entities ending after this timestamp.
        duration_within (bool, optional):
            For entity-level filtering on state/interval entities:
            - True: Entity must be entirely contained within the time range
            - False: Entity can partially overlap with the time range
            Defaults to False. Has no effect on event entities.
        sequence_within (bool, optional):
            For sequence-level filtering (event, interval, state):
            - True: Entire sequence must be contained within the time range
            - False: Sequence can partially overlap with the time range
            Defaults to False.
        date_format (str, optional):
            Custom datetime parsing format for string inputs.

    Examples:
        # Filter entities with partial overlap (default behavior)
        TimeCriterion(
            start_after=datetime.now(),
            end_before=datetime.now() + timedelta(days=90)
        )

        # Only keep state/interval entities fully contained in time range
        TimeCriterion(
            start_after=datetime(2024, 1, 1),
            end_before=datetime(2024, 12, 31),
            duration_within=True
        )

        # Only keep sequences entirely within time range
        TimeCriterion(
            start_after=datetime(2024, 6, 1),
            end_before=datetime(2024, 6, 30),
            sequence_within=True
        )

        # Combine both strict modes
        TimeCriterion(
            start_after=datetime(2024, 1, 1),
            end_before=datetime(2024, 12, 31),
            duration_within=True,
            sequence_within=True
        )
    """

    start_before: Optional[Union[datetime, str, int]] = None
    start_after: Optional[Union[datetime, str, int]] = None
    end_before: Optional[Union[datetime, str, int]] = None
    end_after: Optional[Union[datetime, str, int]] = None
    duration_within: bool = False
    sequence_within: bool = False
    date_format: Optional[str] = None
    # TODO: add settings for duration

    _REGISTER_NAME = "time"
