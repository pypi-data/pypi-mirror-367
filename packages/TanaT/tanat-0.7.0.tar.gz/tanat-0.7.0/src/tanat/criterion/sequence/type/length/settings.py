#!/usr/bin/env python3
"""
Length criterion.
"""

from typing import Optional

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer

from ....base.settings import Criterion
from ....decorator.sequence import sequence_compatible
from .exception import ContradictoryLengthCriterionError


@viewer
@sequence_compatible
@dataclass
class LengthCriterion(Criterion):
    """
    Filter sequences based on their length.

    Provides flexible length-based filtering with multiple comparison
    operators for sequence selection.

    Args:
        gt (int, optional):
            Keep sequences with length strictly greater than this value.
        ge (int, optional):
            Keep sequences with length greater than or equal to this value.
        lt (int, optional):
            Keep sequences with length strictly less than this value.
        le (int, optional):
            Keep sequences with length less than or equal to this value.

    Example:
        # Filter sequences with 10 or fewer events
        LengthCriterion(le=10)
    """

    gt: Optional[int] = None  # greater than
    ge: Optional[int] = None  # greater than or equal to
    lt: Optional[int] = None  # less than
    le: Optional[int] = None  # less than or equal to

    _REGISTER_NAME = "length"

    def __post_init__(self):
        if self.gt is not None and self.ge is not None:
            raise ContradictoryLengthCriterionError(
                "Cannot use 'gt' and 'ge' operators simultaneously"
            )
        if self.lt is not None and self.le is not None:
            raise ContradictoryLengthCriterionError(
                "Cannot use 'lt' and 'le' operators simultaneously"
            )

        lower = self.gt if self.gt is not None else self.ge
        upper = self.lt if self.lt is not None else self.le
        if lower is not None and upper is not None and lower >= upper:
            raise ContradictoryLengthCriterionError(
                f"lower bound ({lower}) must be strictly less than upper bound ({upper})."
            )
