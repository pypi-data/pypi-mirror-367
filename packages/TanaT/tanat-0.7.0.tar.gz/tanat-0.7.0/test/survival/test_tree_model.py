#!/usr/bin/env python3
"""
Tests for Tree survival model.
"""

import pytest

from tanat.survival import SurvivalAnalysis
from tanat.survival.result import SurvivalResult
from tanat.survival.model.type.tree.settings import TreeSurvivalSettings


class TestTreeSurvival:
    """Tests for Tree survival analysis."""

    @pytest.mark.parametrize(
        "pool_type,query",
        [
            ("event", "event_type == 'EXAM'"),
            ("event", "event_type == 'CONSULTATION'"),
            ("state", "health_state == 'CONVALESCENCE'"),
            ("state", "health_state == 'CHRONIC_MONITORING'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_tree_survival_array_generation(
        self, survival_train_pools, pool_type, query, tree_model, snapshot
    ):
        """Test survival array generation with parametrized pool types and queries."""
        sequence_pool = survival_train_pools[pool_type]

        # Generate survival array
        result = tree_model.get_survival_array(sequence_pool=sequence_pool, query=query)

        # Verify result structure
        assert isinstance(result, SurvivalResult)
        assert result.survival_array.dtype.names == ("observed", "duration")

        # Snapshot test on actual result
        snapshot.assert_match(result)

    @pytest.mark.parametrize(
        "pool_type,query",
        [
            ("event", "event_type == 'EXAM'"),
            ("event", "event_type == 'CONSULTATION'"),
            ("state", "health_state == 'CONVALESCENCE'"),
            ("state", "health_state == 'CHRONIC_MONITORING'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_tree_model_fitting_and_structure(
        self, survival_train_pools, pool_type, query, tree_model, snapshot
    ):
        """Test Tree model fitting and survival array structure verification."""
        sequence_pool = survival_train_pools[pool_type]

        # Generate survival array first to check structure
        result = tree_model.get_survival_array(sequence_pool=sequence_pool, query=query)

        # Fit the model
        tree_model.fit(sequence_pool=sequence_pool, query=query)

        # Verify model is fitted
        assert tree_model._model.is_fitted

        # Verify survival array structure
        assert result.survival_array.dtype.names == ("observed", "duration")

        # Snapshot test on survival array directly
        snapshot.assert_match(result.survival_array)

    @pytest.mark.parametrize(
        "pool_type,query",
        [
            ("event", "event_type == 'CONSULTATION'"),
            ("state", "condition == 'DIGESTIVE'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_tree_model_prediction_workflow(
        self,
        survival_train_pools,
        survival_predict_pools,
        pool_type,
        query,
        tree_model,
        snapshot,
    ):
        """Test complete Tree workflow: fit on train, predict on different set."""
        # Use parametrized pool and query for training
        train_pool = survival_train_pools[pool_type]
        predict_pool = survival_predict_pools[pool_type]

        # Generate survival array for training
        _ = tree_model.get_survival_array(sequence_pool=train_pool, query=query)

        # Fit on training data
        tree_model.fit(sequence_pool=train_pool, query=query)

        assert tree_model._model.is_fitted

        survival_functions = tree_model.predict_survival_function(predict_pool)

        # Snapshot test on survival functions directly
        snapshot.assert_match(survival_functions)

    @pytest.mark.parametrize(
        "pool_type,query",
        [
            ("event", "event_type == 'CONSULTATION'"),
            # no way to find a valid workaround
            # ("state", "condition == 'DIGESTIVE'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_tree_model_custom_settings(
        self, survival_train_pools, pool_type, query, snapshot
    ):
        """Test Tree model with custom settings."""
        # Create Tree model with custom settings
        custom_settings = TreeSurvivalSettings(
            max_depth=5,
            min_samples_split=10,
            min_samples_leaf=5,
            max_features="sqrt",
            random_state=123,
        )

        tree_custom = SurvivalAnalysis("tree", settings=custom_settings)

        # Test with parametrized pool
        sequence_pool = survival_train_pools[pool_type]
        result = tree_custom.get_survival_array(
            sequence_pool=sequence_pool, query=query
        )

        assert isinstance(result, SurvivalResult)

        # Test fitting if we have data
        tree_custom.fit(sequence_pool=sequence_pool, query=query)

        assert tree_custom._model.is_fitted

        # Snapshot test on result and fitted status
        snapshot.assert_match(result)
