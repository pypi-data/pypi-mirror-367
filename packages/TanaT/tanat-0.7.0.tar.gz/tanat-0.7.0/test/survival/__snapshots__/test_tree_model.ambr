# serializer version: 1
# name: TestTreeSurvival.test_tree_model_custom_settings[event-event_type == 'CONSULTATION']
  SurvivalResult(survival_array=array([( True, 344.), ( True,  27.), ( True, 178.), ( True, 179.),
         ( True,  59.), ( True,  74.)],
        dtype=[('observed', '?'), ('duration', '<f8')]), sequence_ids=[1, 3, 4, 5, 6, 7], endpoint_dates=1   2023-12-11
  2          NaT
  3   2023-01-28
  4   2023-06-28
  5   2023-06-29
  6   2023-03-01
  7   2023-03-16
  dtype: datetime64[ns], t_zeros=1   2023-01-01
  2   2023-01-01
  3   2023-01-01
  4   2023-01-01
  5   2023-01-01
  6   2023-01-01
  7   2023-01-01
  dtype: datetime64[ns], durations=1    344.0
  3     27.0
  4    178.0
  5    179.0
  6     59.0
  7     74.0
  dtype: float64, observation_data=       start        end  observed
  1 2023-01-01 2023-12-11      True
  3 2023-01-01 2023-01-28      True
  4 2023-01-01 2023-06-28      True
  5 2023-01-01 2023-06-29      True
  6 2023-01-01 2023-03-01      True
  7 2023-01-01 2023-03-16      True)
# ---
# name: TestTreeSurvival.test_tree_model_custom_settings[interval-medication == 'ANTIBIOTIC']
  SurvivalResult(survival_array=array([( True,  19.), ( True,   4.), ( True, 104.), ( True, 144.),
         ( True, 248.), ( True, 137.), ( True,  72.)],
        dtype=[('observed', '?'), ('duration', '<f8')]), sequence_ids=[1, 2, 3, 4, 5, 6, 7], endpoint_dates=1   2023-01-20
  2   2023-01-05
  3   2023-04-15
  4   2023-05-25
  5   2023-09-06
  6   2023-05-18
  7   2023-03-14
  dtype: datetime64[ns], t_zeros=1   2023-01-01
  2   2023-01-01
  3   2023-01-01
  4   2023-01-01
  5   2023-01-01
  6   2023-01-01
  7   2023-01-01
  dtype: datetime64[ns], durations=1     19.0
  2      4.0
  3    104.0
  4    144.0
  5    248.0
  6    137.0
  7     72.0
  dtype: float64, observation_data=       start        end  observed
  1 2023-01-01 2023-01-20      True
  2 2023-01-01 2023-01-05      True
  3 2023-01-01 2023-04-15      True
  4 2023-01-01 2023-05-25      True
  5 2023-01-01 2023-09-06      True
  6 2023-01-01 2023-05-18      True
  7 2023-01-01 2023-03-14      True)
# ---
# name: TestTreeSurvival.test_tree_model_fitting_and_structure[event-event_type == 'CONSULTATION']
  array([( True, 344.), ( True,  27.), ( True, 178.), ( True, 179.),
         ( True,  59.), ( True,  74.)],
        dtype=[('observed', '?'), ('duration', '<f8')])
# ---
# name: TestTreeSurvival.test_tree_model_fitting_and_structure[event-event_type == 'EXAM']
  array([( True,  20.), ( True,  53.), ( True, 230.), ( True, 206.),
         ( True,   1.), ( True, 127.), ( True, 337.)],
        dtype=[('observed', '?'), ('duration', '<f8')])
# ---
# name: TestTreeSurvival.test_tree_model_fitting_and_structure[interval-medication == 'ANTIBIOTIC']
  array([( True,  19.), ( True,   4.), ( True, 104.), ( True, 144.),
         ( True, 248.), ( True, 137.), ( True,  72.)],
        dtype=[('observed', '?'), ('duration', '<f8')])
# ---
# name: TestTreeSurvival.test_tree_model_fitting_and_structure[state-health_state == 'CHRONIC_MONITORING']
  array([( True, 349.), ( True,  94.), ( True,   0.), ( True, 349.)],
        dtype=[('observed', '?'), ('duration', '<f8')])
# ---
# name: TestTreeSurvival.test_tree_model_fitting_and_structure[state-health_state == 'CONVALESCENCE']
  array([( True,   0.), ( True, 158.), ( True,   0.), ( True, 347.)],
        dtype=[('observed', '?'), ('duration', '<f8')])
# ---
# name: TestTreeSurvival.test_tree_model_prediction_workflow[event-event_type == 'CONSULTATION']
  array([StepFunction(x=array([ 27.,  59.,  74., 178., 179., 344.]), y=array([0.66666667, 0.66666667, 0.33333333, 0.33333333, 0.        ,
                0.        ]), a=1.0, b=0.0)                                                                                              ,
         StepFunction(x=array([ 27.,  59.,  74., 178., 179., 344.]), y=array([1.        , 0.66666667, 0.66666667, 0.33333333, 0.33333333,
                0.        ]), a=1.0, b=0.0)                                                                                              ,
         StepFunction(x=array([ 27.,  59.,  74., 178., 179., 344.]), y=array([1.        , 0.66666667, 0.66666667, 0.33333333, 0.33333333,
                0.        ]), a=1.0, b=0.0)                                                                                              ],
        dtype=object)
# ---
# name: TestTreeSurvival.test_tree_model_prediction_workflow[interval-medication == 'ANTIBIOTIC']
  array([StepFunction(x=array([  4.,  19.,  72., 104., 137., 144., 248.]), y=array([0.75, 0.75, 0.5 , 0.25, 0.25, 0.25, 0.  ]), a=1.0, b=0.0),
         StepFunction(x=array([  4.,  19.,  72., 104., 137., 144., 248.]), y=array([1.        , 0.66666667, 0.66666667, 0.66666667, 0.33333333,
                0.        , 0.        ]), a=1.0, b=0.0)                                                                                        ,
         StepFunction(x=array([  4.,  19.,  72., 104., 137., 144., 248.]), y=array([1.        , 0.66666667, 0.66666667, 0.66666667, 0.33333333,
                0.        , 0.        ]), a=1.0, b=0.0)                                                                                        ],
        dtype=object)
# ---
# name: TestTreeSurvival.test_tree_model_prediction_workflow[state-condition == 'DIGESTIVE']
  array([StepFunction(x=array([  0., 198., 199.]), y=array([0.4, 0.2, 0. ]), a=1.0, b=0.0),
         StepFunction(x=array([  0., 198., 199.]), y=array([0.4, 0.2, 0. ]), a=1.0, b=0.0),
         StepFunction(x=array([  0., 198., 199.]), y=array([0.4, 0.2, 0. ]), a=1.0, b=0.0)],
        dtype=object)
# ---
# name: TestTreeSurvival.test_tree_survival_array_generation[event-event_type == 'CONSULTATION']
  SurvivalResult(survival_array=array([( True, 344.), ( True,  27.), ( True, 178.), ( True, 179.),
         ( True,  59.), ( True,  74.)],
        dtype=[('observed', '?'), ('duration', '<f8')]), sequence_ids=[1, 3, 4, 5, 6, 7], endpoint_dates=1   2023-12-11
  2          NaT
  3   2023-01-28
  4   2023-06-28
  5   2023-06-29
  6   2023-03-01
  7   2023-03-16
  dtype: datetime64[ns], t_zeros=1   2023-01-01
  2   2023-01-01
  3   2023-01-01
  4   2023-01-01
  5   2023-01-01
  6   2023-01-01
  7   2023-01-01
  dtype: datetime64[ns], durations=1    344.0
  3     27.0
  4    178.0
  5    179.0
  6     59.0
  7     74.0
  dtype: float64, observation_data=       start        end  observed
  1 2023-01-01 2023-12-11      True
  3 2023-01-01 2023-01-28      True
  4 2023-01-01 2023-06-28      True
  5 2023-01-01 2023-06-29      True
  6 2023-01-01 2023-03-01      True
  7 2023-01-01 2023-03-16      True)
# ---
# name: TestTreeSurvival.test_tree_survival_array_generation[event-event_type == 'EXAM']
  SurvivalResult(survival_array=array([( True,  20.), ( True,  53.), ( True, 230.), ( True, 206.),
         ( True,   1.), ( True, 127.), ( True, 337.)],
        dtype=[('observed', '?'), ('duration', '<f8')]), sequence_ids=[1, 2, 3, 4, 5, 6, 7], endpoint_dates=1   2023-01-21
  2   2023-02-23
  3   2023-08-19
  4   2023-07-26
  5   2023-01-02
  6   2023-05-08
  7   2023-12-04
  dtype: datetime64[ns], t_zeros=1   2023-01-01
  2   2023-01-01
  3   2023-01-01
  4   2023-01-01
  5   2023-01-01
  6   2023-01-01
  7   2023-01-01
  dtype: datetime64[ns], durations=1     20.0
  2     53.0
  3    230.0
  4    206.0
  5      1.0
  6    127.0
  7    337.0
  dtype: float64, observation_data=       start        end  observed
  1 2023-01-01 2023-01-21      True
  2 2023-01-01 2023-02-23      True
  3 2023-01-01 2023-08-19      True
  4 2023-01-01 2023-07-26      True
  5 2023-01-01 2023-01-02      True
  6 2023-01-01 2023-05-08      True
  7 2023-01-01 2023-12-04      True)
# ---
# name: TestTreeSurvival.test_tree_survival_array_generation[interval-medication == 'ANTIBIOTIC']
  SurvivalResult(survival_array=array([( True,  19.), ( True,   4.), ( True, 104.), ( True, 144.),
         ( True, 248.), ( True, 137.), ( True,  72.)],
        dtype=[('observed', '?'), ('duration', '<f8')]), sequence_ids=[1, 2, 3, 4, 5, 6, 7], endpoint_dates=1   2023-01-20
  2   2023-01-05
  3   2023-04-15
  4   2023-05-25
  5   2023-09-06
  6   2023-05-18
  7   2023-03-14
  dtype: datetime64[ns], t_zeros=1   2023-01-01
  2   2023-01-01
  3   2023-01-01
  4   2023-01-01
  5   2023-01-01
  6   2023-01-01
  7   2023-01-01
  dtype: datetime64[ns], durations=1     19.0
  2      4.0
  3    104.0
  4    144.0
  5    248.0
  6    137.0
  7     72.0
  dtype: float64, observation_data=       start        end  observed
  1 2023-01-01 2023-01-20      True
  2 2023-01-01 2023-01-05      True
  3 2023-01-01 2023-04-15      True
  4 2023-01-01 2023-05-25      True
  5 2023-01-01 2023-09-06      True
  6 2023-01-01 2023-05-18      True
  7 2023-01-01 2023-03-14      True)
# ---
# name: TestTreeSurvival.test_tree_survival_array_generation[state-health_state == 'CHRONIC_MONITORING']
  SurvivalResult(survival_array=array([( True, 349.), ( True,  94.), ( True,   0.), ( True, 349.)],
        dtype=[('observed', '?'), ('duration', '<f8')]), sequence_ids=[2, 3, 4, 5], endpoint_dates=1          NaT
  2   2023-12-16
  3   2023-04-05
  4   2023-01-01
  5   2023-12-16
  6          NaT
  7          NaT
  dtype: datetime64[ns], t_zeros=1   2023-01-01
  2   2023-01-01
  3   2023-01-01
  4   2023-01-01
  5   2023-01-01
  6   2023-01-01
  7   2023-01-01
  dtype: datetime64[ns], durations=2    349.0
  3     94.0
  4      0.0
  5    349.0
  dtype: float64, observation_data=       start        end  observed
  2 2023-01-01 2023-12-16      True
  3 2023-01-01 2023-04-05      True
  4 2023-01-01 2023-01-01      True
  5 2023-01-01 2023-12-16      True)
# ---
# name: TestTreeSurvival.test_tree_survival_array_generation[state-health_state == 'CONVALESCENCE']
  SurvivalResult(survival_array=array([( True,   0.), ( True, 158.), ( True,   0.), ( True, 347.)],
        dtype=[('observed', '?'), ('duration', '<f8')]), sequence_ids=[3, 4, 5, 6], endpoint_dates=1          NaT
  2          NaT
  3   2023-01-01
  4   2023-06-08
  5   2023-01-01
  6   2023-12-14
  7          NaT
  dtype: datetime64[ns], t_zeros=1   2023-01-01
  2   2023-01-01
  3   2023-01-01
  4   2023-01-01
  5   2023-01-01
  6   2023-01-01
  7   2023-01-01
  dtype: datetime64[ns], durations=3      0.0
  4    158.0
  5      0.0
  6    347.0
  dtype: float64, observation_data=       start        end  observed
  3 2023-01-01 2023-01-01      True
  4 2023-01-01 2023-06-08      True
  5 2023-01-01 2023-01-01      True
  6 2023-01-01 2023-12-14      True)
# ---
